kong/dao/schemas/plugins_configurations.lua

```diff
@@ -38,6 +38,13 @@ return {
       return false, DaoError("No consumer can be configured for that plugin", constants.DATABASE_ERROR_TYPES.SCHEMA)
     end
 
+    if value_schema.self_check then
+      local ok, err = value_schema.self_check(value_schema, plugin_t.value and plugin_t.value or {}, dao, is_update)
+      if not ok then
+        return false, DaoError(err, constants.DATABASE_ERROR_TYPES.SCHEMA)
+      end
+    end
+
```

Comment:
Sounds good, but you are locking the self_check to a SCHEMA error, and it shouldn't be. It can be any other error type too. You should return the error as it is returned by `self_check`, it is up to the plugin to decide what error it is exactly. Maybe the error comes from Cassandra being unaccessible for example.

Otherwise the check sounds good to me. A better solution would be to investigate on why it is not invoked from `validate_entity`, because the function calls itself on sub-schemas so if there is no other error, `self_check` should be recursively called already.